ClientApp/src/components/FileUploader.js

```diff
@@ -0,0 +1,103 @@
+import AWS from 'aws-sdk';
+import { useState } from 'react';
+import {Input, Image, Button, Form} from 'semantic-ui-react'
+
+AWS.config.update({
+  accessKeyId: '<ACCESS-KEY-ID>',
+  secretAccessKey: '<SECRET-ACCESS-KEY>',
+  region: 'us-east-1',
+  signatureVersion: 'v4',
+});
+
+const FileUploader = () =>  {
+    const s3 = new AWS.S3();
+    const [fileUrl, setFileUrl] = useState(null);
+    const [file, setFile] = useState(null);
+    const [typeId, setTypeId] = useState(null);
+    const [englishName, setEnglishName] = useState(null);
+    const [copticName, setCopticName] = useState(null);
+    const [isAddedToDb, setAddedToDb] = useState(false);
+  
+    const handleFileSelect = (e) => {
+      setFile(e.target.files[0]);
+    }
+
+    const handleTypeId = (e) => {
+      setTypeId(e.target.value);
+    }
+
+    const handleEnglishName = (e) => {
+      setEnglishName(e.target.value);
+    }
+
+    const handleCopticName = (e) => {
+      setCopticName(e.target.value);
+    }
+
+    const upload = async () => {
+        if (!file) {
+          return;
+        }
+
+        const params = { 
+          Bucket: 'test-bucket-hymns', 
+          Key: `${Date.now()}.${file.name}`, 
+          Body: file 
+        };
+        const { Location } = await s3.upload(params).promise();
+        setFileUrl(Location);
+        console.log('uploading to s3', Location);
+
+        const asset = { 
+          AssetId: 0,
+          TypeId: typeId, 
+          Location: Location,
+          EnglishName: englishName,
+          CopticName: copticName
+        };
+        
+        const response = await fetch('asset', {
```

Comment:
how does fetch know where to 'fetch' from?

Reply:
It uses a method called `createProxyMiddleware` to create a proxy to the backend. You can find a this logic in `ClientApp/src/setupProxy.js`.

Here's a quick read on general usage of proxy in react: https://create-react-app.dev/docs/proxying-api-requests-in-development/ if you're interested!